.github/workflows/build-ui-and-server.yml

```diff
@@ -195,12 +195,12 @@ jobs:
         run: npx playwright test --project=chromium --project=firefox --reporter="./tests/e2e/custom-playwright-reporter.js"
         working-directory: ui
       - name: Save PR number
-        if: always()
+        if: ${{ !cancelled() }}
```

Comment:
This one is better than always() because we can cancel the workflow anytime

Reply:
Oh, nice. I wasn't aware of this.